modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js

```diff
@@ -404,20 +464,30 @@ const Select = ({
 				onSetActive={setExpand}
 				ref={menuElementRef}
 			>
-				<ClayDropDown.ItemList>
-					{options.map((option, index) => (
-						<DropdownItem
-							currentValue={currentValue}
-							expand={expand}
-							index={index}
-							key={`${option.value}-${index}`}
-							multiple={multiple}
-							onSelect={handleSelect}
-							option={option}
-							options={options}
-						/>
-					))}
-				</ClayDropDown.ItemList>
+				{options.length > MAX_ITEMS ? (
+					<DropdownWithSearch
+						currentValue={currentValue}
+						expand={expand}
+						handleSelect={handleSelect}
+						multiple={multiple}
+						options={options}
+					/>
+				) : (
+					<ClayDropDown.ItemList>
+						{options.map((option, index) => (
+							<DropdownItem
+								currentValue={currentValue}
+								expand={expand}
+								index={index}
+								key={`${option.value}-${index}`}
+								multiple={multiple}
+								onSelect={handleSelect}
+								option={option}
+								options={options}
+							/>
+						))}
+					</ClayDropDown.ItemList>
```

Comment:
@alinedoleron, once this same component is being used inside the component `DropdownWithSearch`, can we try to change the logic to re-use the invocation of the component `ClayDropDown.ItemList`?